File: frontend/lib/models/conversation.dart
class Conversation {
  final String id;
  final String title;
  final String? lastMessage;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Conversation({
    required this.id,
    required this.title,
    this.lastMessage,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Conversation.fromJson(Map<String, dynamic> json) => Conversation(
    id: json['id'] as String,
    title: json['title'] as String,
    lastMessage: json['last_message'] as String?,
    createdAt: DateTime.parse(json['created_at'] as String),
    updatedAt: DateTime.parse(json['updated_at'] as String),
  );
}

File: backend/app/schemas/conversation.py
from pydantic import BaseModel
from datetime import datetime


class ConversationItem(BaseModel):
    id: str
    title: str
    last_message: str | None = None
    created_at: datetime
    updated_at: datetime


class MessageItem(BaseModel):
    id: str
    role: str
    type: str
    content: str
    audio_url: str
    created_at: datetime


class UpdateTitleRequest(BaseModel):
    title: str
